app/src/main/java/com/vultisig/wallet/presenter/chain_coin/ChainCoinViewmodel.kt

```diff
@@ -0,0 +1,28 @@
+package com.vultisig.wallet.presenter.chain_coin
+
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+import com.google.gson.Gson
+import com.vultisig.wallet.models.Coin
+import com.vultisig.wallet.models.Vault
+import com.vultisig.wallet.models.getBalanceInFiat
+import com.vultisig.wallet.ui.models.ChainAccountUiModel
+import com.vultisig.wallet.ui.navigation.Screen
+import dagger.hilt.android.lifecycle.HiltViewModel
+import java.math.BigDecimal
+import java.math.RoundingMode
+import javax.inject.Inject
+
+@HiltViewModel
+class ChainCoinViewmodel @Inject constructor(savedStateHandle: SavedStateHandle) : ViewModel() {
+    private val gson = Gson()
+    private val accountJson: String = savedStateHandle.get<String>(Screen.CHAIN_COIN_ACCOUNT_JSON)!!
```

Comment:
I don't think this is the correct way to pass vault , instead , we should just pass the vault's pubkeyECDSA , and also the selected chain. 
1. Vault can be retrieved from VaultDB , all the coins has been saved in vault
2. Chain name , which represent which chain current view is for.